File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(result, count, limit):
        nonlocal answer
        if count == limit:
            if result == target:
                answer += 1
            return
        
        dfs(result + numbers[count], count + 1, limit)
        dfs(result - numbers[count], count + 1, limit)
    
    dfs(0, 0, len(numbers))
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    
    const dfs = (curr, result) => {
        // 종료 조건
        if (curr === numbers.length) {
            if (result === target) answer++;
            
            return;
        }
        
        dfs(curr + 1, result + numbers[curr]);
        dfs(curr + 1, result - numbers[curr]);
    }
    
    dfs(0, 0)
    
    return answer;
}